### ISSUE TITLE
Property consolidation

### ISSUE THREAD

**Issue by tbranyen** [grommet/grommet#2529]
There are many unique property names to Grommet that could be aligned better with the web specification. For instance the property `a11yTitle` is odd since it has numbers and is an abbreviation of accessibility. I would expect this property to be named `ariaLabel`.

Other examples of uniquely named props:

- `gap` vs `margin`
- `pad` vs `padding`
- `tag` vs `as`

A suggestion would be to consolidate the properties over the course of two full major version changes. In v3 introduce deprecation warnings for using the old props with suggestions for the new ones, possibly produce a codemod to do this for your users, and in v4 completely remove the old props.

**Comment by oorestisime** [comment]
the `tag` vs `as` part is already at the consolidation stage. in case you found out a component that supports `tag` but not `as` shout and i could handle it as well!

**Comment by tbranyen** [comment]
@oorestisime awesome, where can I find out more about the consolidation effort? I don't see any indication in the code that `tag` is being deprecated.

If `tag` and `as` are getting aligned that's great, but what about the other props I listed?

**Comment by oorestisime** [comment]
The issue is: https://github.com/grommet/grommet/issues/2487
There is no warning when a user uses `tag` right now. But there is a Note in the documentation https://github.com/grommet/grommet/blob/master/src/js/components/Box/doc.js#L240-L242
There should be the same Note in the other components, unless i missed some.

Can't speak for the other props, i just know about the `tag` and `as` since i worked recently  on them.

**Comment by ericsoderberghp** [comment]
Box currently supports both `margin` and `gap`. How were you envisioning moving `gap` to `margin`?

**Comment by ericsoderberghp** [comment]
`pad` was a legacy from grommet v1, partly due to trying to keep things shorter and simpler. It hasn't been an issue before that it isn't called `padding`. We have had questions about `a11tTitle`.

**Comment by tbranyen** [comment]
@ericsoderberghp I'll dig more into `gap`, on the surface it looks to be very similar to `margin`.

**Comment by alansouzati** [comment]
I agree with deprecating `a11yTitle` and rely on `aria-label` instead.

**Comment by episodeyang** [comment]
The type for this is currently set to `String` by mistake. Shall we change the type to accommodate custom components the same way that the styled components do it?

#2869

**Comment by powerglove-eth** [comment]
I like gap and pad.
